Data/ByteString/Lazy.hs

```diff
@@ -460,6 +462,12 @@ foldr :: (Word8 -> a -> a) -> a -> ByteString -> a
 foldr k = foldrChunks (flip (S.foldr k))
 {-# INLINE foldr #-}
 
+-- | 'foldr'' is like 'foldr', but strict in the accumulator.
+foldr' :: (Word8 -> a -> a) -> a -> ByteString -> a
+foldr' _ a Empty = a
+foldr' f a (Chunk c cs) = S.foldr' f (foldr' f a cs) c
```

Comment:
Please factor out `foldr' f a = go where go ...`, similar to `foldl'` above. Otherwise GHC is unlikely to figure out that these arguments are shared by all recursive calls and do not need to be passed over and over again.

Reply:
Done.